tests/samplers_tests/tpe_tests/test_multivariate_parzen_estomator.py

```diff
@@ -0,0 +1,148 @@
+        "f": None,
+    }
+    categorical_weights = {
+        "a": None,
+        "b": None,
+        "c": None,
+        "d": None,
+        "e": None,
+        "f": np.array([[0.25, 0.5, 0.25]] + consider_prior * [[1.0 / 3.0, 1.0 / 3.0, 1.0 / 3.0]]),
+    }
+
+    for param_name, values in mpe._sigmas.items():
+        assert np.all(
+            np.equal(mpe._sigmas[param_name], sigmas[param_name])
+        ), 'parameter "{}"'.format(param_name)
+        assert np.all(np.equal(mpe._mus[param_name], mus[param_name])), "parameter: {}".format(
+            param_name
+        )
+        assert np.all(
+            np.equal(mpe._categorical_weights[param_name], categorical_weights[param_name])
+        ), "parameter: {}".format(param_name)
+
+
+def test_sample_multivariate_parzen_estimator() -> None:
+
+    parameters = _ParzenEstimatorParameters(
+        consider_prior=False,
+        prior_weight=0.0,
+        consider_magic_clip=False,
+        consider_endpoints=False,
+        weights=lambda x: np.arange(x) + 1.0,
+    )
+
+    with patch(_precompute_sigmas0, return_value=1e-8 * np.ones(2)):
+        mpe = _MultivariateParzenEstimator(multivariate_samples, search_space, parameters)
+
+    # Test the shape of the samples.
+    output_multivariate_samples = mpe.sample(np.random.RandomState(0), 3)
+    for param_name in output_multivariate_samples.keys():
+        assert output_multivariate_samples[param_name].shape == (3,)
+
+    # Test the values of the output.
+    # As we set ``consider_prior`` = False and pre-computed sigma to be 1e-8,
+    # the samples almost equals to the input ``multivariate_samples``.
+    output_multivariate_samples = mpe.sample(np.random.RandomState(0), 1)
+    for param_name, samples in output_multivariate_samples.items():
+        if samples.dtype == str:
+            assert samples[0] == "y", "parameter {}".format(param_name)
+        else:
+            assert np.allclose(samples, multivariate_samples[param_name]), "parameter {}".format(
+                param_name
+            )
+
+    # Test the output when the seeds are fixed.
+    assert output_multivariate_samples == mpe.sample(np.random.RandomState(0), 1)
+
+
+def test_log_pdf_multivariate_parzen_estimator() -> None:
+
+    parameters = _ParzenEstimatorParameters(
+        consider_prior=False,
+        prior_weight=1.0,
+        consider_magic_clip=True,
+        consider_endpoints=True,
+        weights=lambda x: np.arange(x) + 1.0,
+    )
+    # Parzen estimator almost becomes mixture of Dirac measures.
+    with patch(_precompute_sigmas0, return_value=1e-8 * np.ones(1)):
+        mpe = _MultivariateParzenEstimator(multivariate_samples, search_space, parameters)
+
+    log_pdf = mpe.log_pdf(multivariate_samples)
+    output_multivariate_samples = mpe.sample(np.random.RandomState(0), 100)
+    output_log_pdf = mpe.log_pdf(output_multivariate_samples)
+    # The more data points, the smaller the value of ``log_pdf`` should be.
```

Comment:
I am not sure about the second comment (sorry, it's probably because of my unclear code). As can be found in the first comment, the log-likelihoods of the random samples (`output_multivaraite_samples`) are w.h.p. less than the samples used for the construction of the Parzen estimator (`multivariate_samples`) because the Parzen estimator becomes almost becomes a mixture of Dirac measures. For such Dirac measures, we can expect that the log-likelihood of the points becomes almost zero except for centers of each Dirac measures.

Reply:
Thanks for the comment. I understood the theoretical reason for that. I will change the comment as follow: "The likelihood of the previous observations is a positive value, and that of the points sampled by the Parzen estimator is zero with high probability." What do you think?

Reply:
Yes, that sounds OK to me but a small suggestion; I guess "almost" zero is a more accurate expression.

Reply:
Thanks for the comment. I will change so soon.